lib/storage/providers/SQLiteStorage.js

```diff
@@ -0,0 +1,118 @@
+    * @param {String[]} keys
+    * @return {Promise<Array<[key, value]>>}
+    */
+    multiGet(keys) {
+        const placeholders = _.map(keys, () => '?').join(',');
+        const command = `SELECT record_key, valueJSON FROM keyvaluepairs WHERE record_key IN (${placeholders});`;
+        return db.executeAsync(command, keys)
+            .then(({rows}) => {
+                // eslint-disable-next-line no-underscore-dangle
+                const result = _.map(rows._array, row => [row.record_key, JSON.parse(row.valueJSON)]);
+                return result;
+            });
+    },
+
+    /**
+    * Sets the value for a given key. The only requirement is that the value should be serializable to JSON string
+    * @param {String} key
+    * @param {*} value
+    * @return {Promise<void>}
+    */
+    setItem(key, value) {
+        return db.executeAsync('REPLACE INTO keyvaluepairs (record_key, valueJSON) VALUES (?, ?);', [key, JSON.stringify(value)]);
+    },
+
+    /**
+    * Stores multiple key-value pairs in a batch
+    * @param {Array<[key, value]>} pairs
+    * @return {Promise<void>}
+    */
+    multiSet(pairs) {
+        const stringifiedPairs = _.map(pairs, pair => [
+            pair[0],
+            JSON.stringify(pair[1]),
+        ]);
+        return db.executeBatchAsync([['REPLACE INTO keyvaluepairs (record_key, valueJSON) VALUES (?, json(?));', stringifiedPairs]]);
+    },
+
+    /**
+    * Multiple merging of existing and new values in a batch
+    * @param {Array<[key, value]>} pairs
+    * @return {Promise<void>}
+    */
+    multiMerge(pairs) {
+        // Note: We use `ON CONFLICT DO UPDATE` here instead of `INSERT OR REPLACE INTO`
+        // so the new JSON value is merged into the old one if there's an existing value
+        const query = `INSERT INTO keyvaluepairs (record_key, valueJSON) 
+             VALUES (:key, JSON(:value)) 
+             ON CONFLICT DO UPDATE 
+             SET valueJSON = JSON_PATCH(valueJSON, JSON(:value)); 
+        `;
+        const queryArguments = _.map(pairs, (pair) => {
+            const value = JSON.stringify(pair[1]);
+            return [pair[0], value];
+        });
+        return db.executeBatchAsync([[query, queryArguments]]);
+    },
+
+    /**
+    * Returns all keys available in storage
+    * @returns {Promise<String[]>}
+    */
+    getAllKeys: () => db.executeAsync('SELECT record_key FROM keyvaluepairs;').then(({rows}) => {
+        // eslint-disable-next-line no-underscore-dangle
+        const result = _.map(rows._array, row => row.record_key);
+        return result;
+    }),
+
+    /**
+    * Removes given key and it's value from storage
+    * @param {String} key
+    * @returns {Promise<void>}
+    */
+    removeItem: key => db.executeAsync('DELETE FROM keyvaluepairs WHERE record_key = ?;', [key]),
+
+    /**
+    * Clears absolutely everything from storage
+    * @returns {Promise<void>}
+    */
+    clear: () => db.executeAsync('DELETE FROM keyvaluepairs;', []),
```

Comment:
We ran into an issue where clearing with `multiSet` [here](https://github.com/Expensify/react-native-onyx/blob/c26f6e5defdeb8183ff205053a20bde943b6f5d1/lib/Onyx.js#L1061) in Onyx caused iOS to crash because we were setting keys to `null`. 

I was looking through the AsyncStorage docs and saw there is a [`multiRemove`](https://reactnative.dev/docs/asyncstorage#multiremove) which I think would work better in this situation.

For SQLite I believe it would be as simple as:
```JS
`DELETE FROM keyvaluepairs WHERE record_key IN (${keysToRemove.join(',')});`;
```

Thoughts?

Reply:
#expensify-open-source [thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1671583867011199) for context.

Reply:
Sounds good! Do you want me to add that method?

Reply:
If no one else is opposed 🙂

Reply:
I don't have any strong feelings about why this shouldn't exist - but looks like it would need to be implemented in the [web storage](https://github.com/Expensify/react-native-onyx/blob/main/lib/storage/providers/LocalForage.js) as well.

Reply:
Maybe this could be a follow up given we need to add it to both providers

Reply:
I think, since it's going to be a smaller change to rework how `clear` works it can be a PR that is merged and tested before the current PR 

But I'm also OK moving forward with this PR the way it is, as long as we're addressing the `clear` problem in a separate PR

Reply:
yep, agree with @kidroca and @mountiny I think this PR has been open for long enough and we can do a follow up to look at improving `clear()`.